eth/client.go

```diff
@@ -418,7 +419,174 @@ func (c *client) Bond(amount *big.Int, to ethcommon.Address) (*types.Transaction
 		}
 	}
 
-	return c.BondingManagerSession.Bond(amount, to)
+	// Get transcoder pool
+	transcoders, err := c.TranscoderPool()
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get transcoder pool")
+	}
+
+	// Get max pool size
+	maxSize, err := c.GetTranscoderPoolMaxSize()
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get transcoder pool max size")
+	}
+
+	// Get delegator
+	delegator, err := c.GetDelegator(sender)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get delegator")
+	}
+
+	isFull := int64(len(transcoders)) == maxSize.Int64()
+
+	// Switching delegate's calculate old delegate positions
+	var oldHints lpTypes.TranscoderPoolHints
+	if delegator.DelegateAddress != to {
```

Comment:
We can add `delegator.DelegateAddress != ethcommon.Address{}` to this conditional to skip the code block below when `Bond()` is called by an account that has not staked yet.

Reply:
fixed in 13e5e37